airflow/contrib/hooks/gcp_translate_hook.py

```diff
@@ -0,0 +1,103 @@
+
+from google.cloud.translate_v2 import Client
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+
+
+class CloudTranslateHook(GoogleCloudBaseHook):
+    """
+    Hook for Google Cloud translate APIs.
+    """
+
+    _client = None
+
+    def __init__(self, gcp_conn_id='google_cloud_default'):
+        super(CloudTranslateHook, self).__init__(gcp_conn_id)
+
+    def get_conn(self):
+        """
+        Retrieves connection to Cloud Translate
+
+        :return: Google Cloud Translate client object.
+        :rtype: Client
+        """
+        if not self._client:
+            self._client = Client(credentials=self._get_credentials())
+        return self._client
+
+    def translate(
+        self, values, target_language, format_=None, source_language=None, customization_ids=(), model=None
+    ):
+        """Translate a string or list of strings.
+
+        See https://cloud.google.com/translate/docs/translating-text
+
+        :type values: str or list
+        :param values: String or list of strings to translate.
+
+        :type target_language: str
+        :param target_language: The language to translate results into. This
+                                is required by the API and defaults to
+                                the target language of the current instance.
+
+        :type format_: str
+        :param format_: (Optional) One of ``text`` or ``html``, to specify
+                        if the input text is plain text or HTML.
+
+        :type source_language: str or None
+        :param source_language: (Optional) The language of the text to
+                                be translated.
+
+        :type customization_ids: str or list or None
+        :param customization_ids: (Optional) ID or list of customization IDs
+                                  for translation. Sets the ``cid`` parameter
+                                  in the query.
+
+        :type model: str or None
+        :param model: (Optional) The model used to translate the text, such
+                      as ``'base'`` or ``'nmt'``.
+
+        :rtype: str or list
+        :returns: A list of dictionaries for each queried value. Each
+                  dictionary typically contains three keys (though not
+                  all will be present in all cases)
+
+                  * ``detectedSourceLanguage``: The detected language (as an
+                    ISO 639-1 language code) of the text.
+                  * ``translatedText``: The translation of the text into the
+                    target language.
+                  * ``input``: The corresponding input value.
+                  * ``model``: The model used to translate the text.
+
+                  If only a single value is passed, then only a single
+                  dictionary will be returned.
+        :raises: :class:`~exceptions.ValueError` if the number of
+                 values and translations differ.
+        """
+        client = self.get_conn()
+
```

Comment:
Should we add some log for debug easy?

Reply:
We already have a debug log in the Operator. https://github.com/apache/airflow/pull/4755/files#diff-250e99f89f67f9e59efc842ed614fb59R116 which I used to see the results. And it makes sense to put it there usually - especially that sometimes several different hooks are used in single "execute" method of the operator and we usually add some logging there (as needed).

Reply:
Ok, I get this.